cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -553,9 +541,6 @@ public void getStatuses(HttpRequest request, HttpResponder responder,
         ProgramStatus programStatus = lifecycleService.getProgramStatus(programId);
         statuses.add(new BatchProgramStatus(
           program, HttpResponseStatus.OK.getCode(), null, programStatus.name()));
-      } catch (BadRequestException e) {
-        statuses.add(new BatchProgramStatus(
```

Comment:
Ohh I see `getProgramStatus` does not throw `BadRequestException` and also it does not make sense to catch `BadRequestException` it in loop.